src/libpmem2/badblocks.h

```diff
@@ -26,6 +26,19 @@ extern "C" {
 #define BB_NOT_SUPP \
 	"checking bad blocks is not supported on this OS, please switch off the CHECK_BAD_BLOCKS compat feature using 'pmempool-feature'"
 
+struct pmem2_badblock_iterator {
+	int ndctl_has_namespace_badblock_iterator;
+	int n_iter; /* number of iterations */
+
+	/* needed only by the ndctl namespace badblock iterator */
+	struct ndctl_namespace *ndns;
+
+	/* needed only by the ndctl region badblock iterator */
+	struct ndctl_region *region;
+	unsigned long long ns_beg;
+	unsigned long long ns_end;
```

Comment:
IIUC either ndns is not null or region is not null (and ns_* are set)
I think it makes sense to group region data in a struct and either use a union or replace ndctl_has_namespace_badblock_iterator by ndns != NULL check.

Reply:
Done